apps/opik-frontend/src/lib/utils.ts

```diff
@@ -23,6 +24,23 @@ export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
 
+export const isStringMarkdown = (string: unknown) => {
+  if (isString(string)) {
+    const markdownPatterns = [
+      /#\s\w+/, // headers
+      /\*{2}\w+\*{2}/, // bold
+      /\*\w+\*/, // italics
+      /\[.+\]\(.+\)/, // links
+      /!\[.*\]\(.*\)/, // images
+      /\d\.\s\w+/, // numbered lists
+      /\*\s\w+/, // bullet lists
+    ];
+
+    return markdownPatterns.some((pattern) => pattern.test(string));
+  }
+  return false;
+};
```

Comment:
smart things. maybe you could also add checks for  inline code and code blocks, horizontal rules `---` and tables